src/ImageSharp.Drawing/Processing/Processors/Drawing/FillRegionProcessor.cs

```diff
@@ -168,17 +168,31 @@ protected override void OnFrameApply(ImageFrame<TPixel> source, Rectangle source
                         {
                             if (!this.Options.Antialias)
                             {
+                                bool hasOnes = false;
+                                bool hasZeros = false;
                                 for (int x = 0; x < scanlineWidth; x++)
                                 {
                                     if (scanline[x] >= 0.5)
                                     {
                                         scanline[x] = 1;
+                                        hasOnes = true;
                                     }
                                     else
                                     {
                                         scanline[x] = 0;
+                                        hasZeros = true;
                                     }
                                 }
+
+                                if (hasOnes != hasZeros && this.IsSolidBrushWithoutBlending(out SolidBrush<TPixel> solidBrush))
```

Comment:
`this.IsSolidBrushWithoutBlending(out SolidBrush<TPixel> solidBrush)` should be only done once outside the loops as it can't change between lines

Reply:
Good point, thanks!  I'll update it.

Reply:
Sorry it took a while (I don't work on this project often), the requested changes are now done!